File: app/src/main/java/htw/gma_sose22/metronompro/ui/metronome/MetronomeViewModel.java
package htw.gma_sose22.metronompro.ui.metronome;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class MetronomeViewModel extends ViewModel {

    private final MutableLiveData<String> mText;

    public MetronomeViewModel() {
        mText = new MutableLiveData<>();
        mText.setValue("This is dashboard fragment");
    }

    public LiveData<String> getText() {
        return mText;
    }

}

File: app/src/main/java/htw/gma_sose22/metronompro/ui/metronome/MetronomeViewModel.kt
package htw.gma_sose22.metronompro.ui.metronome

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import htw.gma_sose22.metronomprokit.metronome.Metronome
import htw.gma_sose22.metronomprokit.metronome.MetronomeService

class MetronomeViewModel : ViewModel() {

    private val mutableBPM: MutableLiveData<Int> = MutableLiveData()

    val bpm: LiveData<Int>
        get() = mutableBPM

    init {
        mutableBPM.value = Metronome.DEFAULT_SPEED
    }

    fun handleBPMChangeRequested(bpmDelta: Int) {
        MetronomeService.changeBPM(bpmDelta)
        mutableBPM.value = MetronomeService.bpm
    }

    fun handleStartStopButtonClicked() {
        MetronomeService.togglePlayback()
    }
}